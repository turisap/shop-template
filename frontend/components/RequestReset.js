import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';
import { CURRENT_USER_QUERY } from "./User";

const RESET_REQUEST_MUTATION = gql`
    mutation RESET_REQUEST_MUTATION($email : String!) {
        requestReset(email : $email) {
            message
        }
    }
`;


class SignIn extends Component {
    state = {
        email : ""
    };

    saveToState = e => {
        this.setState({
            [e.target.name] : e.target.value
        })
    };

    render() {
        return (
            <Mutation
                mutation={RESET_REQUEST_MUTATION}
                variables={this.state}
            >
                {(resetFunction, {error, loading, called}) => {
                    return (
                        <Form
                            method="post"
                            onSubmit={async e => {
                                e.preventDefault();
                                await resetFunction();
                                this.setState({email : ""});
                            }}
                        >
                            <fieldset disabled={loading} aria-busy={loading}>
                                <h2>Request a password reset</h2>
                                <Error error={error}/>
                                {!error && !loading && called && <p>You reset request has been successful, check your email</p>}
                                <label>
                                    Email
                                    <input
                                        type="email"
                                        name="email"
                                        placeholder="Email"
                                        value={this.state.email}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <button type="submit">Request reset</button>
                            </fieldset>
                        </Form>
                    )
                }}
            </Mutation>
        );
    }
}

SignIn.propTypes = {};

export default SignIn;
