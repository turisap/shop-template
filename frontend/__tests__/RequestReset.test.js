import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import RequestReset, { RESET_REQUEST_MUTATION } from '../components/RequestReset';
import { MockedProvider } from 'react-apollo/test-utils';

const mocks = [
    {
        request : {
                query : RESET_REQUEST_MUTATION,
                variables : { email : 'k@mail.com'}
        },
        result  : {
            data : {
                requestReset : {
                    message : 'success',
                    __typename : 'SuccessMessage'
                }
            }
        }
    }
];

describe('<RequestReset/>', () => {
    it('renders and matches snapshot', async () => {
        const wrapper = mount(
            <MockedProvider>
                <RequestReset/>
            </MockedProvider>
        );
        //console.log(wrapper.debug())
        const form = wrapper.find('[data-test="reset-form"]');
        //console.log(form.debug())
        expect(toJSON(form)).toMatchSnapshot();
    })

    // it('calls the mutation', async () => {
    //     const wrapper = mount(
    //         <MockedProvider mocks={mocks}>
    //             <RequestReset/>
    //         </MockedProvider>
    //     );
    //     // simulate typing email
    //     wrapper
    //         .find('input')
    //         .simulate('change', { target : { name : 'email', value : 'k@mail.com '}});
    //     // submit the form
    //     wrapper
    //         .find('form')
    //         .simulate('submit');
    //     await wait();
    //     wrapper.update();
    //     console.log(wrapper.debug())
    //     //expect(wrapper.find('p')).toContain('alksdfj');
    // })
})
