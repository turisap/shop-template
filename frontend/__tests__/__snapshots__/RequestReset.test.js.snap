// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
Array [
  <Form
    data-test="reset-form"
    method="post"
    onSubmit={[Function]}
  >
    <form
      className="Form-sc-1ns280c-0 fTwjpq"
      data-test="reset-form"
      method="post"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Request a password reset
        </h2>
        <DisplayError
          error={Object {}}
        />
        <label>
          Email
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Request reset
        </button>
      </fieldset>
    </form>
  </Form>,
  <form
    className="Form-sc-1ns280c-0 fTwjpq"
    data-test="reset-form"
    method="post"
    onSubmit={[Function]}
  >
    <fieldset
      aria-busy={false}
      disabled={false}
    >
      <h2>
        Request a password reset
      </h2>
      <DisplayError
        error={Object {}}
      />
      <label>
        Email
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Request reset
      </button>
    </fieldset>
  </form>,
]
`;
